ase/api/src/enum.c

```diff
@@ -781,18 +781,34 @@ fpgaPropertiesSetBBSVersion(fpga_properties prop, fpga_version bbs_version)
 fpga_result __FPGA_API__
 fpgaPropertiesGetVendorID(const fpga_properties prop, uint16_t *vendor_id)
 {
-	UNUSED_PARAM(prop);
-	UNUSED_PARAM(vendor_id);
-	FPGA_MSG("Vendor ID not supported");
-	return FPGA_NOT_SUPPORTED;
+	struct _fpga_properties *_prop = (struct _fpga_properties *) prop;
+	fpga_result result = FPGA_INVALID_PARAM;
+	if (NULL == _prop || NULL == vendor_id) {
+		FPGA_ERR("Attempting to dereference NULL pointer(s)");
+		return result;
+	}
+	if (FIELD_VALID(_prop, FPGA_PROPERTY_VENDORID)) {
```

Comment:
Do you also need to check if (_prop->magic != FPGA_PROPERTY_MAGIC) ? I see this check in fpgaPropertiesGetDeviceID().